iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -257,7 +258,8 @@ public Class<? extends AbstractSerDe> getSerDeClass() {
   public HiveMetaHook getMetaHook() {
     // Make sure to always return a new instance here, as HiveIcebergMetaHook might hold state relevant for the
     // operation.
-    return new HiveIcebergMetaHook(conf);
+    return StringUtils.isEmpty(MetastoreUtil.getCatalogType(conf)) ?
```

Comment:
we should check this. what hook would be used in case of hadoop catalog, see hadoop_catalog_create_table.q